source/parse.h

```diff
@@ -4389,6 +4428,26 @@ struct translation_unit_node
     }
 };
 
+// Definitions of out-of-line dtors for nodes with unique_ptr members of forward-declared types
```

Comment:
I've decided to add the deferred dtor definitions here, at the point where (to the best of my knowledge) all node types are already defined.